### PR TITLE
Json test report, UI fixes, unit tests

### PR DESCRIPTION
**Pull request by kkateq**
1. Fixed UI issues with json test report
2. Added unit tests.

### CODE REVIEW

File: www/script/project/rtTestReport.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                for (i = 0; i < settings.length; i += 1) {
+                    settings[i].oInstance.fnFilter(sInput, iColumn, bRegex, bSmart);
+                }
+
+                $('.dataTables_empty').closest(th).hide();
+            };
+
+            //Filter when return is hit
+            jQuery.fn.dataTableExt.oApi.fnFilterOnReturn = function () {
+                var that = this;
```
Comment (opdude) [inline www/script/project/rtTestReport.js]: Why do we want to have that and this equal the same thing?

File: www/script/project/rtTestReport.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+                this.each(function (i) {
+                    $.fn.dataTableExt.iApiIndex = i;
+                    var anControl = $('input', that.fnSettings().aanFeatures.f);
+                    anControl.unbind('keyup').bind('keypress.katana', function (e) {
+                        if (e.which === 13) {
```
Comment (opdude) [inline www/script/project/rtTestReport.js]: Maybe we should mention that keypress 13 is enter for those who don't remember it by heart. We could even have it as a global setting

File: www/script/project/rtTestReport.js (hunk 1 above)
Comment (kkateq) [inline www/script/project/rtTestReport.js]: I do not know :) I've not reviewed this logic - just moved private methods to public - to be able to unit test new logic

File: www/script/project/rtTestReport.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        setupFilterButtons: function () {
+            // submit on return
+            var $filterInput = $("#filterinput"),
+                $submitButton = $('#submitFilter'),
+                failedTests = $(".log-main").attr("data-failed-tests");
+
+            $filterInput.keydown(function (event) {
+                // Filter on the column (the index) of this element
+                var e = window.event || event;
+                if (e.keyCode === 13) {
```
Comment (opdude) [inline www/script/project/rtTestReport.js]: Again mention what 13 means

File: www/script/test/test_realtimeRouting.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        rtGlobal = require("rtGlobal");
+
+
+    describe("Add page init handler", function () {
+        it("page init handler has been executed", function (done) {
+            var page_id = "test_handler";
+            realtimerouting.addPageInitHandler(page_id, done);
+
+            helpers.getCurrentPage = function() { return page_id; }
+            realtimerouting.init();
```
Comment (opdude) [inline www/script/test/test_realtimeRouting.js]: You didn't add any expects?

File: www/script/test/test_realtimeRouting.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        });
+
+        it("rtGlobal init handler has been executed", function (done) {
+            rtGlobal.init  = done;
+
+            helpers.getCurrentPage = function() { return ""; }
+            realtimerouting.init();
```
Comment (opdude) [inline www/script/test/test_realtimeRouting.js]: Same here

File: www/script/test/test_realtimeRouting.js (hunk 4 above)
Comment (kkateq) [inline www/script/test/test_realtimeRouting.js]: So the 'done' handler here should be invoked,  if it is not invoked - during realtimerouting.init() call - the test will fail(it's jasmine test callback)

File: www/script/test/test_realtimeRouting.js (hunk 5 above)
Comment (kkateq) [inline www/script/test/test_realtimeRouting.js]: The same here - I expect that 'done' is invoked - otherwise - test will fail

**Comment by kkateq** [comment]
Fixed. Please review again

**Comment by opdude** [comment]
Looks good :)

**Comment by kkateq** [comment]
Additional UI fixes after report testing

**Comment by opdude** [comment]
Can you take a look at the loop.index value with jinja :)? Otherwise it's fine

**Comment by kkateq** [comment]
Fixed.

**Pull request merged**